core/src/main/java/io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java

```diff
@@ -0,0 +1,812 @@
+        private int rebalanceToThreadId;
+        private String rebalanceTableName;
+        private boolean rebalanceReleasedByFromThread;
+
+        private LineTcpMeasurementEvent(int maxMeasurementSize, MicrosecondClock clock, LineProtoTimestampAdapter timestampAdapter) {
+            lexer = new TruncatedLineProtoLexer(maxMeasurementSize);
+            cache = lexer.getCharSequenceCache();
+            this.clock = clock;
+            this.timestampAdapter = timestampAdapter;
+            lexer.withParser(new LineProtoParser() {
+                @Override
+                public void onLineEnd(CharSequenceCache cache) {
+                }
+
+                @Override
+                public void onEvent(CachedCharSequence token, int type, CharSequenceCache cache) {
+                    assert cache == LineTcpMeasurementEvent.this.cache;
+                    switch (type) {
+                        case EVT_MEASUREMENT:
+                            assert measurementNameAddress == 0;
+                            measurementNameAddress = token.getCacheAddress();
+                            break;
+
+                        case EVT_TAG_NAME:
+                        case EVT_TAG_VALUE:
+                            assert firstFieldIndex == -1;
+                            addresses.add(token.getCacheAddress());
+                            break;
+                        case EVT_FIELD_NAME:
+                            if (firstFieldIndex == -1) {
+                                firstFieldIndex = addresses.size() / 2;
+                            }
+                        case EVT_FIELD_VALUE:
+                            assert firstFieldIndex != -1;
+                            addresses.add(token.getCacheAddress());
+                            break;
+                        case EVT_TIMESTAMP:
+                            assert timestampAddress == 0;
+                            timestampAddress = token.getCacheAddress();
+                            break;
+
+                        default:
+                            throw new RuntimeException("Unrecognised type " + type);
+
+                    }
+                }
+
+                @Override
+                public void onError(int position, int state, int code) {
+                    assert errorPosition == -1;
+                    errorPosition = position;
+                    errorCode = code;
+                }
+            });
+        }
+
+        long parseLine(long bytesPtr, long hi) {
+            clear();
+            long recvBufLineNext = lexer.parseLine(bytesPtr, hi);
+            if (recvBufLineNext != -1) {
+                if (!isError() && firstFieldIndex == -1) {
+                    errorPosition = (int) (recvBufLineNext - bytesPtr);
+                    errorCode = LineProtoParser.ERROR_EMPTY;
+                }
+
+                if (!isError()) {
+                    if (timestampAddress == 0) {
+                        timestamp = clock.getTicks();
```

Comment:
This is strange line, I can see below that `timestampAddress` does get set to 0, but this line is never triggered by any of the tests. Can we review what's happening here?

Reply:
This happens when the measurement does not include a timestamp and we use the received time.
I have added the following test to cover it
LineTcpConnectionContextTest#testUseReceivedTimestamp1